src/components/streamSelect/StreamSelect.js

```diff
@@ -114,6 +127,27 @@ class StreamSelect extends Component {
             <button>Play</button>
           </form>
         </div>
+
+        <div className="latency-select-wrap">
+          <form
+            className="latency-select"
+            onSubmit={(e) => {
```

Comment:
Is this normal for React? I'd expect a set method to take a value, not an event as argument. And I'd expect an update method to take no value argument, but update the state based on instance/global state.

How about something like this instead: `onSubmit={this.handleTargetLatencySubmit}`
And below on the input: `onChange={this.handleTargetLatencyChange}`

What are the general conventions in React? I based this on: https://reactjs.org/docs/handling-events.html

Reply:
Because I'm using an ES6 class to create a React component, the .this scope is not available in your example. 
To fix this I can either use arrow functions to call my methods, or use `.bind(this)` on every method in the constructor (Or use createReactClass, which doesn't use ES6 and is apparently worse for performance). I think arrow functions are probably the cleanest solution in this case.

I'm using this documentation as a reference: https://reactjs.org/docs/react-without-es6.html#autobinding

Reply:
Ah, of course. The whole thing just looks messy and overly complicated. What's the use of React here at all really? Why not just use the input to store the value? E.g: get rid of customLatencyField state and just refer to the targetLatencyInput.value instead? Same for customStreamID. That way you can set the values based on query parameters on page load and you will only have to listen to submit events of their forms.

Reply:
I changed the latency input to not use onChange anymore. Like discussed it instead now checks for the value in the submit event. But I'm not able to use this approach with the stream ID input field. This is because when a custom ID is used in the URL,  I set the value of the input field to that ID using state. Because of React I also need an onChange event (documentation suggests this as well), otherwise the user won't be able to change the value. I agree that doing it this way is a bit messy but I think avoiding this by using refs or accessing the DOM directly is also messy. Let me know what you think